pyccel/ast/numpyext.py

```diff
@@ -173,6 +175,67 @@ def _sympystr(self, printer):
     def arg(self):
         return self._arg
 
+#==============================================================================
+class NumpyArange(NumpyNewArray):
```

Comment:
Could this inherit from `PythonRange` and `NumpyNewArray` to avoid re-implementing some of the `PythonRange` properties? Or does this create some kind of problematic diamond shaped class inheritance?

Reply:
hmm, I don't think it can, at least if the PythonRange class is changed just as you did in your PR

Reply:
Why not? I would have thought that would help if anything?

Reply:
I'm not that good at inheritance, but pythonRange property returns elements stored in self._args, and in arange i don't do that.